EIPS/eip-7557.md

```diff
@@ -0,0 +1,370 @@
+        "accessors": [
+          {
+            "sender": "0x90F8bf6A479f320ead074411a4B0e7944Ea8c9C1",
+            "priorityFeePerGas": "1000"
+          },
+          {
+            "sender": "0x90F8bf6A479f320ead074411a4B0e7944Ea8c9C1",
+            "priorityFeePerGas": "2000"
+          },
+          {
+            "sender": "0x22d491Bde2303f2f43325b2108D26f1eAbA1e32b",
+            "priorityFeePerGas": "2000"
+          },
+          {
+            "sender": "0x22d491Bde2303f2f43325b2108D26f1eAbA1e32b",
+            "priorityFeePerGas": "3000"
+          }
+        ]
+      },
+      {
+        "id": "0x0000000000000000000000000000000000000000000000000000000000000007",
+        "accessors": [
+          {
+            "sender": "0xFFcf8FDEE72ac11b5c542428B35EEF5769C409f0",
+            "priorityFeePerGas": "1000"
+          },
+          {
+            "sender": "0xFFcf8FDEE72ac11b5c542428B35EEF5769C409f0",
+            "priorityFeePerGas": "2000"
+          }
+        ]
+      }
+    ]
+  }
+]
+
+```
+
+### Calculating a reimbursement of the burned base fee
+
+Considering that the same amount of computation is needed to access an address or a slot regardless of the number of
+transactions using one, it is reasonable for the protocol to only burn the gas cost of the cold access once.
+As all transactions in the same block pay exactly the same `baseFeePerGas`, the single cost of accessing a cold item is
+divided evenly among all transactions containing such access and the rest of the burned base fee is reimbursed.
+
+### Setting an absolute minimal cost of cold state access
+
+If a large number of transactions all access the same addresses or slots, the cost of each cold access may get
+way too low which may represent a potential DoS attack vector.
+
+In order to prevent that, the gas cost of including an entity in the access list cannot be lower
+than `MIN_ACCESS_LIST_ENTRY_COST`, which is set to `32 gas`.
+
+This value is equivalent to the calldata cost of including two bytes long identifier of entries in `block_access_list`.
+
+### Calculating a reimbursement of the charged priority fee
+
+Each transaction pays an individual `priorityFeePerGas` value and redistributing this part of the cold access cost
+is more complex.\
+We propose the following approach to a fair reimbursement of the paid `priorityFee`:
+
+1. The validator gets paid the `priorityFee` for each cold access only once, but according to the highest `priorityFee`
+   among the transactions containing the said cold access.
+2. The rest of the Ether that was charged by the validator as a  `priorityFee` is redistributed back to all the senders
+   of transactions containing the same cold access in proportion to their **marginal contribution** to the total reimbursement.
+3. The `marginal contribution` of a transaction to the total reimbursement is defined as the difference between the sum total
```

Comment:
so the contribution of the the most expensive transaction is the second largest priority fee * gas cost. although this is fine but cost could just be proportionately reimbursed keeping it simple